client/runtime/src/main/java/io/quarkiverse/openapi/generator/AuthConfig.java

```diff
@@ -47,48 +41,91 @@ public class AuthConfig {
      * @see SpecItemConfig
      * @see OpenApiGeneratorConfig
      */
-    @ConfigItem
-    public Optional<String> headerName;
+    Optional<String> headerName();
 
     /**
-     * Configures a particular parameter value to be used by any of the different internal authentication filters
-     * that processes the different securityScheme definitions.
-     * <p>
+     * Sets the Basic Authentication username for a given OpenAPI securityScheme.
+     * <p/>
      * For example, given a file named petstore.json with a securityScheme named "petstore-basic-auth", that is of
-     * http basic authentication type, the following configuration can establish the user and password to be used.
-     * must be used.
-     * <p>
+     * http basic authentication type, the following configuration can establish the user to be used.
+     * <p/>
      * quarkus.openapi-generator.petstore_json.auth.petstore_basic_auth.username=MyUserName
-     * quarkus.openapi-generator.petstore_json.auth.petstore_basic_auth.password=MyPassword
      *
-     * @see AuthsConfig
-     * @see SpecItemConfig
-     * @see OpenApiGeneratorConfig
-     * @see BasicAuthenticationProvider
-     * @see BearerAuthenticationProvider
-     * @see ApiKeyAuthenticationProvider
+     * @return the username portion for Basic Authentication
+     * @see AuthConfig#password()
+     * @see <a href="https://spec.openapis.org/oas/v3.1.0.html#basic-authentication-sample">4.8.27.2.1 Basic Authentication
+     *      Sample</a>
      */
-    @ConfigItem(name = ConfigItem.PARENT)
-    public Map<String, String> authConfigParams;
```

Comment:
Is `@WithParentName` not necessary here?

Reply:
Why? This implementation was lazy at the time, leaving it as a can-be-anything string. Now we have settled the properties in the API so that the birds can chirp again.

Reply:
Oh you mean in `username`? No, we don't need it. The IDE with Quarkus plugin installed now recognizes these properties beautifully.